core/src/main/java/nl/nn/adapterframework/webcontrol/api/TestServiceListener.java

```diff
@@ -81,24 +81,23 @@ public Response getServiceListeners() throws ApiException {
 	@Path("/test-servicelistener")
 	@Produces(MediaType.APPLICATION_JSON)
 	@Consumes(MediaType.MULTIPART_FORM_DATA)
-	public Response postServiceListeners(MultipartFormDataInput input) throws ApiException {
+	public Response postServiceListeners(MultipartBody inputDataMap) throws ApiException {
 		Map<String, Object> result = new HashMap<String, Object>();
 
 		String message = null, serviceName = null, dispatchResult = null;
 		InputStream file = null;
 
-		Map<String, List<InputPart>> inputDataMap = input.getFormDataMap();
 		try {
-			if(inputDataMap.get("message") != null)
-				message = inputDataMap.get("message").get(0).getBodyAsString();
-			if(inputDataMap.get("service") != null) {
-				serviceName = inputDataMap.get("service").get(0).getBodyAsString();
+			if(inputDataMap.getAttachmentObject("message", String.class) != null)
+				message = inputDataMap.getAttachmentObject("message", String.class);
+			if(inputDataMap.getAttachmentObject("service", String.class) != null) {
```

Comment:
resolveStringFromMap(inputDataMap, "service");

Reply:
do you want to remove the "if"...? because resolveStringFromMap never return null.